iothub_client/tests/message_queue_ut/message_queue_ut.c

```diff
@@ -1603,4 +1603,191 @@ TEST_FUNCTION(do_work_in_progress_queue_timeout)
     message_queue_destroy(mq);
 }
 
+// Tests_SRS_MESSAGE_QUEUE_21_070: [The message_queue_move_all_back_to_pending shall add all in_progress message in front of the pending messages.]
```

Comment:
>add all in_progress message in front of the pending messages [](start = 85, length = 60)

might be overkill but do we have a test which makes sure this is true (the ordering of the messages)

Reply:
Short answer is no.

There is no good way to test it. We should be able to inspect the queue orders, and there is no API for that.

---
In reply to: [333112087](https://github.com/Azure/azure-iot-sdk-c/pull/1244#discussion_r333112087) [](ancestors = 333112087)